File: old/src/classes/galaxy.cpp
// galaxy.cpp
// 2014-10-12T23:27:26-07:00

#include "../pch.h"
#include "galaxy.h"

namespace FoxTrader
{
    Galaxy::Galaxy(Universe *a_universe, uint32_t a_index)
    {
        this->m_universe = a_universe;
        this->m_index = a_index;

        this->m_name = Tools::GenerateName();
        this->m_position = vec3_t::Random(kGalaxySizeMax);

        /* Law of attraction */
        this->m_systems.reserve(kSystemsMax);
        for (uint32_t i = 0; i < kSystemsMax; ++i)
        {
            this->m_systems.push_back(System(this, i));
        }
    }

    Galaxy::~Galaxy()
    {
        this->m_universe = NULL;
        this->m_systems.clear();
    }

    std::vector<System>* Galaxy::GetSystems()
    {
        return &this->m_systems;
    }

    uint32_t Galaxy::GetIndex()
    {
        return this->m_index;
    }

    std::string Galaxy::GetName()
    {
        return this->m_name;
    }

    void Galaxy::Tick()
    {
        for (uint32_t i = 0; i < kSystemsMax; ++i)
        {
            this->m_systems[i].Tick();
        }
    }
}

File: Game/Galaxy.cs
﻿using System.Collections.Generic;
using FoxTrader.Game.Utils;

namespace FoxTrader.Game
{
    class Galaxy : ITickable
    {
        private Universe m_universe;
        private readonly int m_index;

        private readonly string m_name;

        private readonly Vector3 m_position;

        private readonly List<System> m_systems;

        public Galaxy(Universe c_universe, int c_index)
        {
            m_universe = c_universe;
            m_index = c_index;

            m_name = Generator.Name();
            m_position = Vector3.Random(Constants.kGalaxySizeMax);

            m_systems = new List<System>(Constants.kSystemsMax);

            for (var a_idx = 0; a_idx <= Constants.kSystemsMax; a_idx++)
            {
                m_systems.Add(new System(this, a_idx));
            }
        }

        public Vector3 Position => m_position;

        public int Index => m_index;

        public string Name => m_name;

        public void Tick()
        {
            m_systems.ForEach(c_system => c_system.Tick());
        }
    }
}
